artichoke-backend/src/state/mod.rs

```diff
@@ -32,6 +34,7 @@ pub struct State {
     pub parser: Option<parser::State>,
     pub classes: Registry<class::Spec>,
     pub modules: Registry<module::Spec>,
+    pub encodings: HashSet<encoding::Spec>,
```

Comment:
I think a better design is to pass encodings around as symbols (i.e. a `u32`) both in the interpreter storage, `mrb_value` repr for `Encoding`s, and the flags of a `String` `mrb_value`.

If we end up doing that, we should use an `FxHashSet` here from `rustc_hash` which will perform much better